google/cloud/opentelemetry/quickstart/README.md

```diff
@@ -0,0 +1,201 @@
+# HOWTO: Google Cloud C++ Client Library + Cloud Trace integration.
+
+This directory contains a small example showing how to enable OpenTelemetry
+tracing for a Google Cloud Storage (GCS) client. The traces are collected and
+sent to Cloud Trace.
+
+This quickstart assumes the reader is familiar with how to build and use the GCS
+client. If not, see the [GCS quickstart](/google/cloud/storage/quickstart).
+
+The quickstart also assumes the reader is familiar with how to build the Cloud
+Trace C++ client library. If not, see the
+[Cloud Trace quickstart](/google/cloud/trace/quickstart). (Note that the Cloud
+Trace Exporter requires dependencies on gRPC and Protobuf, which the GCS client
+does not).
+
+## The Quickstart
+
+### What it does
+
+The quickstart installs a Cloud Trace [exporter]. It creates a GCS client, with
+tracing enabled. The client makes some routine calls, each of which are traced.
+Upon exit of the quickstart, the collected traces are sent to Cloud Trace.
+
+### Viewing the trace in Cloud Trace
+
+To find the traces created by this quickstart, we can follow the documentation
+to [Find and view traces]. We can open the Cloud Trace UI in the Google Cloud
+project we supplied to the quickstart. We can search using the filter
+`SpanName:storage::Client` to find any spans created by the GCS client.
+
+![google_cloud_cpp_opentelemetry_quickstart](https://user-images.githubusercontent.com/23088558/235268195-c19bf903-7358-4a0f-ac72-438b5286463b.png)
+
+For an overview of the Cloud Trace UI, see: [View traces overview].
+
+## OpenTelemetry Dependency
+
+The Google Cloud C++ client libraries use [OpenTelemetry] to provide
+observability into their operation at runtime. To enable tracing,
+`google-cloud-cpp` takes a direct dependency on
+[`opentelemetry-cpp`][opentelemetry-cpp].
+
+OpenTelemetry provides full [install instructions][opentelemetry-cpp-install].
+We will highlight the important points.
+
+### Compatibility with Abseil
+
+OpenTelemetry backports Abseil types, defining them in `namespace absl`. The
+Google Cloud client libraries depends on Abseil, which also defines these types
+in the same namespace. In order to avoid ambiguous symbols, we must set certain
+flags when compiling `opentelemetry-cpp`.
+
+### Testing
+
+The client library is only tested against the latest version of
+`opentelemetry-cpp`. That is v1.9.0, circa 2023-05.
+
+## Using with Bazel
+
+> :warning: If you are using Windows or macOS there are additional instructions
+> at the end of this document.
+
+1. Install Bazel using [the instructions][bazel-install] from the `bazel.build`
+   website.
+
+1. Compile this example using Bazel:
+
+   ```bash
+   cd $HOME/google-cloud-cpp/google/cloud/opentelemetry/quickstart
+   bazel build ...
```

Comment:
I think we should show the magic `--@io_opentelemetry//...` and `--@google_cloud_cpp//` flags here.  These commands do not work otherwise, right?